### PR TITLE
Update PlayerResources capacity from INotify* methods

### PR DESCRIPTION
**Pull request by teinarss**
Some refactoring on how StoreResources capacity updates  PlayerResources.ResourceCapacity

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/StoresResources.cs [hunk 1]
```diff
@@ -49,7 +49,10 @@ public StoresResources(Actor self, StoresResourcesInfo info)
 
 		void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
 		{
+			var capacity = info.Capacity;
+			player.RemoveStorage(capacity);
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/StoresResources.cs]: In theory the game already removes an actor from the world during owner change (see [Actor.cs#L328](https://github.com/OpenRA/OpenRA/blob/824db72a4cfbd8c6f82bbe60b4bea5c6da327f56/OpenRA.Game/Actor.cs#L328)), so this should already be done from inside the `INotifyRemovedFromWorld.RemovedFromWorld` call or am I missing something here?

Comment (teinarss) [reply OpenRA.Mods.Common/Traits/StoresResources.cs]: Looks like you are right, based this pattern on how its implemented in Power trait.

**Comment by teinarss** [comment]
Updated

**Review by abcdefg30 — APPROVED** [review]

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
The branch hasn't been rebased in a while, but I checked locally that rebasing doesn't cause any hidden conflicts or regressions, so lgtm.

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/422df0279973daea626efae863a01436d6de481c...408214481a51f760a813e750467a4e01e1adaf99

**Pull request merged**